.github/workflows/python-package.yml

```diff
@@ -2,39 +2,46 @@ name: Python package
 
 on:
   push:
-    branches: [ master ]
+    branches: [master]
   pull_request:
     branches:
       - master
-      - 'dev*'
-      - 'release*'
+      - "dev*"
+      - "release*"
 
 jobs:
   build:
-
     runs-on: ubuntu-latest
     strategy:
       matrix:
-        python-version: ['3.8', '3.9', '3.10', '3.11']
+        python-version: ["3.8", "3.9", "3.10", "3.11"]
 
     steps:
-    - uses: actions/checkout@v2
-    - name: Set up Python ${{ matrix.python-version }}
-      uses: actions/setup-python@v2
-      with:
-        python-version: ${{ matrix.python-version }}
-    - name: Install dependencies
-      run: |
-        python -m pip install --upgrade pip
-        if [ -f dev-requirements.txt ]; then pip install -r dev-requirements.txt; fi
-    - name: Test with pytest
-      run: |
-        pytest --pycodestyle -v --cov --cov-report=xml
-    - name: Upload code coverage to Codecov
-      if: ${{ matrix.python-version==3.8 }}
-      uses: codecov/codecov-action@v3
-      with:
-        token: ${{ secrets.CODECOV_TOKEN }} # not required for public repos
-        file: ./coverage.xml
-        name: codecov-umbrella
-        fail_ci_if_error: true
+      - uses: actions/checkout@v2
+      - name: Set up Python ${{ matrix.python-version }}
+        uses: actions/setup-python@v2
+        with:
+          python-version: ${{ matrix.python-version }}
+      - name: Lint with black
+        uses: psf/black@stable
+        with:
+          options: "--check --diff --line-length=79"
```

Comment:
A make target to run Lint and formatting would seem to be better.

For example, define `make lint` in the Makefile in the same way as `make format`. Of course, run by IDEs would be modern. However, I often just use make.

Reply:
I've just added a `make lint` command in favor of `make format` (I suppose one always wants to perform all linting). I don't see a conflict with having both the `make` command and the github actions step, as the second one validates linting for all commits (also by those people that didn't look at the Makefile, such as myself). If you agree, feel free to merge. If not, just let me know what to change!